frontend/lib/identification/activation_workflow/activation_code_scanner_page.dart

```diff
@@ -107,15 +109,24 @@ class ActivationCodeScannerPage extends StatelessWidget {
           throw const ActivationInvalidTotpSecretException();
         }
         final totpSecret = const Base64Decoder().convert(activationResult.totpSecret!);
-        debugPrint('Card Activation: Successfully activated.');
 
-        provider.setCode(DynamicUserCode()
+        DynamicUserCode userCode = DynamicUserCode()
           ..info = activationCode.info
           ..pepper = activationCode.pepper
           ..totpSecret = totpSecret
           ..cardVerification = (CardVerification()
             ..cardValid = true
-            ..verificationTimeStamp = secondsSinceEpoch(DateTime.parse(activationResult.activationTimeStamp))));
+            ..verificationTimeStamp = secondsSinceEpoch(DateTime.parse(activationResult.activationTimeStamp)));
+        if (isAlreadyInList(userCodesModel.userCodes, userCode)) {
```

Comment:
I think this check needs to move to the overwriteExisting state. As otherwise you're first prompet whether you want to overwrite an existing card and after that you get the message that the card already exists.

Reply:
This would also break the state of the card on the current device.

Reply:
Alright i moved it and also send only the `CardInfo` to the function, since the rest of the `DynamicUserCode` won't be checked